tools/src/main/scala/scala/scalanative/optimizer/pass/AllocLowering.scala

```diff
@@ -21,8 +23,15 @@ class AllocLowering(implicit fresh: Fresh, top: Top) extends Pass {
 
     insts.foreach {
       case Let(n, Op.Classalloc(ClassRef(cls))) =>
-        val size = let(Op.Sizeof(cls.layout.struct))
-        let(n, Op.Call(allocSig, alloc, Seq(cls.rtti.const, size), Next.None))
+        val size = MemoryLayout.sizeOf(cls.layout.struct)
+        val allocMethod =
+          if (size < LARGE_OBJECT_MIN_SIZE) alloc else largeAlloc
```

Comment:
@densh Here we need the check to take into account the header of the object in the GC. The total allocated size needs to be less than 8K. But the header size can change depending on the GC. Should we add GC specific properties in compiler ?

If the compiler would know about the GC, we could also adapt the injection passes.